src/main/java/redis/clients/jedis/XReadArgs.java

```diff
@@ -2,40 +2,96 @@
 
 import redis.clients.jedis.exceptions.JedisDataException;
 
-import java.util.LinkedList;
-import java.util.List;
+import java.util.*;
 
 public class XReadArgs {
 
-    public final List<String> streams;
-    public final List<String> offsets;
-    public final int count;
-    public final boolean block;
-    public final long blockDuration;
+    public final List<String> streamsAndOffsets;
+    public int count;
+    public boolean block;
+    public long blockDuration;
+
+    /**
+     * Creates simple XReadArgs object.
+     * @param streamsAndOffsets List of stream names and offsets... stream names first, offsets second.
+     */
+    public XReadArgs(List<String> streamsAndOffsets) {
+        this.streamsAndOffsets = streamsAndOffsets;
+        this.count = 0;
+        this.block = false;
+        this.blockDuration = 0;
+    }
 
-    private XReadArgs(List<String> streams, List<String> offsets, int count, boolean block, long blockDuration) {
-        this.streams = streams;
-        this.offsets = offsets;
+    private XReadArgs(List<String> streamsAndOffsets, int count, boolean block, long blockDuration) {
+        this.streamsAndOffsets = streamsAndOffsets;
         this.count = count;
         this.block = block;
         this.blockDuration = blockDuration;
     }
 
+    public void setStreamsAndOffsets(List<String> streamsAndOffsets) {
+        this.streamsAndOffsets.clear();
+        if (streamsAndOffsets.size() < 50) {
```

Comment:
Is there any need to arbitrarily  call add all after size is larger than 50? Might as well just call add no matter the size unless I am missing something...

Reply:
just wanted to take advantage of System.arrayCopy in situations where the list is super long.  Trading an allocation for performance when using `addAll`